tests/exception/thandle_in_finally.nim

```diff
@@ -0,0 +1,33 @@
+discard """
+  description: '''
+    Ensure that a finally clause can fully handle an exception by re-raising
+    and catching it
+  '''
+  knownIssue.c js vm: '''
+    Not implemented properly by the code generator and/or runtime
+  '''
+"""
+
+# XXX: it's questionable whether this really should work / be allowed
```

Comment:
I'm thinking this through myself:

- on line 17 we decide to 'cancel the current computation', targeting the handler on line 29
- on line 20 we decide to 'retarget the cancellation handler', now on line 21
- upon leaving the line 21 cancellation handler, we decide to continue the rest of the computation

Reading through my characterization, I think the 'precise' question is, should a `finally` be allowed re-parent an existing exception (cancellation)? I _believe_ the answer is "yes, it should" because if not then, the property we'd be after is that `finally` cannot `raise`, which seems infeasible/not very useful. I consider this a variant of raising an entirely new exception that overrides the old one, except in this case our 'new exception' 'adopts' the data of the old one.

Reply:
Hm, I'm not sure I completely follow. A `finally` could be allowed to raise without having to allow re-targeting an existing exception to a handler within the `finally` (e.g., by disallowing re-raise outside of `except` branches.

In your model, would line 28 be executed?

Reply:
> In your model, would line 28 be executed?

That's a great question, I want to say yes. But that would mean we entered the `finally` through exceptional means, it handled the exception internally, then the treatment would become non-exceptional.

Which seems to defer the determination of how to leave a `finally` that much more delayed. Where each control flow branch within a `finally` can change the handling... ooph.

Perhaps a static restriction is a better bet, because we can always lift the restriction later when/if we're willing to do that extra analysis and think through all the implications (extra accumulated experience in the future will also be beneficial).

Reply:
> Which seems to defer the determination of how to leave a `finally` that much more. Where each control flow branch within a `finally` can change the handling... ooph.

Yeah, this is the reason why I'd prefer not supporting the re-targeting.

> Perhaps a static restriction is a better bet, because we can always lift the restriction later when/if we're willing to do that extra analysis and think through all the implications (extra accumulated experience in the future will also be beneficial).

Okay, in that case I would make a follow up that statically disallows a re-raise outside of `except` branches.

Reply:
Awesome, and thanks for working through the reasoning with me.